File: cognitive-exercises-frontend/src/app/shared/model/user-score-dto.ts
import {PortfolioDto} from './portfolio-dto';

export class UserScoreDto {
  public place: number;
  public username: string;
  public portfolio: PortfolioDto

  constructor(place: number, username: string, portfolio: PortfolioDto) {
    this.place = place;
    this.username = username;
    this.portfolio = portfolio;
  }
}

File: cognitive-exercises-data/src/main/java/com/zelazobeton/cognitiveexercises/model/UserScoreDto.java
package com.zelazobeton.cognitiveexercises.model;

import com.zelazobeton.cognitiveexercises.domain.Portfolio;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class UserScoreDto {
    private Integer place;
    private String username;
    private PortfolioDto portfolio;

    public UserScoreDto(Portfolio portfolio, int place) {
        this.place = place;
        this.username = portfolio.getUser().getUsername();
        this.portfolio = new PortfolioDto(portfolio);
    }
}
